shudder/queue.py

```diff
@@ -18,54 +18,103 @@
 
 """
 import json
-
-from boto.utils import get_instance_metadata
-import boto.sns as sns
-import boto.sqs as sqs
+import boto3
+import hashlib
 
 from shudder.config import CONFIG
+import shudder.metadata as metadata
 
 
-INSTANCE_ID = get_instance_metadata()['instance-id']
+INSTANCE_ID = metadata.get_instance_id()
 QUEUE_NAME = "{prefix}-{id}".format(prefix=CONFIG['sqs_prefix'],
                                     id=INSTANCE_ID)
 
 
 def create_queue():
-    """Creates the SQS queue and returns the connection/queue"""
-    conn = sqs.connect_to_region(CONFIG['region'])
-    queue = conn.create_queue(QUEUE_NAME)
-    queue.set_timeout(60 * 60)  # one hour
+    """Creates the SQS queue and returns the queue url and metadata"""
+    conn = boto3.client('sqs', region_name=CONFIG['region'])
+    queue_metadata = conn.create_queue(QueueName=QUEUE_NAME, Attributes={'VisibilityTimeout':'3600'})
+    """Get the SQS queue object from the queue URL"""
+    sqs = boto3.resource('sqs', region_name=CONFIG['region'])
+    queue = sqs.Queue(queue_metadata['QueueUrl'])
     return conn, queue
 
 
 def subscribe_sns(queue):
+    """Attach a policy to allow incoming connections from SNS"""
```

Comment:
ER MER GERD.

A good chunk of this is pulled straight from boto2. See if there's a policy allowing incoming messages from SNS. To do this we need a unique policy ID. Then make sure it looks how we need it to looks. Which is basically to ONLY accept SNS pushes from the SNS topic we've subscribed to.